infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -124,43 +124,43 @@ public LogReplicationSinkManager(String localCorfuEndpoint, LogReplicationConfig
                 .keyStore((String) context.getServerConfig().get(ConfigParamNames.KEY_STORE))
                 .ksPasswordFile((String) context.getServerConfig().get(ConfigParamNames.KEY_STORE_PASS_FILE))
                 .tlsEnabled((Boolean) context.getServerConfig().get("--enable-tls"))
-                .maxCacheEntries(configManager.getConfig().getMaxCacheSize())
+                .maxCacheEntries(replicationContext.getConfig().getMaxCacheSize())
                 .maxWriteSize(context.getMaxWriteSize())
                 .build())
                 .parseConfigurationString(localCorfuEndpoint).connect();
         this.pluginConfigFilePath = context.getPluginConfigFilePath();
         this.topologyConfigId = topologyConfigId;
-        init(metadataManager, configManager, replicationSession);
+        init(metadataManager, replicationContext, replicationSession);
     }
 
     /**
      * Constructor Sink Manager
      *
      * @param localCorfuEndpoint endpoint for local corfu server
-     * @param configManager log replication configuration manager
+     * @param replicationContext log replication context which provides configuration
      */
     @VisibleForTesting
-    public LogReplicationSinkManager(String localCorfuEndpoint, LogReplicationConfigManager configManager,
+    public LogReplicationSinkManager(String localCorfuEndpoint, LogReplicationContext replicationContext,
                                      LogReplicationMetadataManager metadataManager, String pluginConfigFilePath,
                                      ReplicationSession replicationSession) {
         this.runtime =  CorfuRuntime.fromParameters(CorfuRuntime.CorfuRuntimeParameters.builder()
-                .maxCacheEntries(configManager.getConfig().getMaxCacheSize()).build())
+                .maxCacheEntries(replicationContext.getConfig().getMaxCacheSize()).build())
```

Comment:
minor: It will be a bit cleaner if configManager has the getConfig() method instead of context IMO

Reply:
Maybe let's quickly discuss these 3 comments offline and decide

Reply:
I think something similar, by keeping the "context.getXXX()" pattern it might be cleaner and easier to maintain.

As for the configManager initialization part, I'm a bit inclined to do it in current way (init in discoveryService, pass to context, and context will be just a wrapper) . as we are using `ServerContext` to init configManager, so we will have to pass `ServerContext` to `LogReplicationContext` if we init configManager there, which changes the `LogReplicationContext` interface as well.    but not strong preference and open to ideas

Reply:
I am fine with both kinds of interfaces - context.getConfig() or context.getConfigManager().getConfig().

Whichever we decide, let us keep the usage uniform because right now there are both usages.